usaspending_api/search/v2/views/spending_by_award.py

```diff
@@ -0,0 +1,276 @@
+            return Response(self.populate_response(results=[], has_next=False))
+
+    def create_class_attributes(self, filters, subawards):
+        if subawards:
+            self.populate_attributes_for_subawards(filters)
+        else:
+            self.populate_attributes_for_awards(filters)
+
+    def construct_queryset(self):
+        queryset = self.custom_queryset_order_by(self.base_queryset, self.sort_by_fields, self.pagination["sort_order"])
+        queryset = queryset.values(*list(self.database_fields))
+        return queryset[self.pagination["lower_bound"] : self.pagination["upper_bound"]]
+
+    def create_response(self, queryset):
+        has_next = len(queryset) > self.pagination["limit"]
+        results = []
+        for record in queryset[: self.pagination["limit"]]:
+            row = {"internal_id": record[self.internal_id_field]}
+
+            for field in self.fields:
+                row[field] = record.get(self.type_code_to_field_map[record[self.award_semaphore]].get(field))
+
+            if "Award ID" in self.fields:
+                for id_type in self.award_id_fields:
+                    if record[id_type]:
+                        row["Award ID"] = record[id_type]
+                        break
+            results.append(row)
+
+        return self.populate_response(results=results, has_next=has_next)
+
+    def populate_attributes_for_subawards(self, filters):
+        """ sub-award logic for SpendingByAward"""
+        minimum_db_fields = {"subaward_number", "piid", "fain", "award_type"}
+        api_to_db_mapping_list = [contract_subaward_mapping, grant_subaward_mapping]
+
+        self.base_queryset = subaward_filter(filters)
+        self.award_semaphore = "award_type"
+        self.award_id_fields = ["award__piid", "award__fain"]
+        self.internal_id_field = "subaward_number"
+        self.database_fields = self.get_database_fields(minimum_db_fields, api_to_db_mapping_list)
+        self.type_code_to_field_map = {"procurement": contract_subaward_mapping, "grant": grant_subaward_mapping}
+
+        if self.pagination["sort_key"] == "Award ID":
+            self.sort_by_fields = self.award_id_fields
+        elif set(filters["award_type_codes"]) <= set(procurement_type_mapping):
+            self.sort_by_fields = [contract_subaward_mapping[self.pagination["sort_key"]]]
+        elif set(filters["award_type_codes"]) <= set(assistance_type_mapping):
+            self.sort_by_fields = [grant_subaward_mapping[self.pagination["sort_key"]]]
+
+    def populate_attributes_for_awards(self, filters):
+        """ Award logic for SpendingByAward"""
+        minimum_db_fields = {"award_id", "piid", "fain", "uri", "type"}
+        model = obtain_view_from_award_group(filters["award_type_codes"])
+        api_to_db_mapping_list = [
+            award_contracts_mapping,
+            award_idv_mapping,
+            loan_award_mapping,
+            non_loan_assistance_award_mapping,
+        ]
```

Comment:
Does this need to be a function level variable? Seems like a constant to me. Maybe even something we could have in the mappings file/folder itself as an "all mappings" so that one isn't missed someday.